### PR TITLE
Add safety dependency to alloc and std crates

### PR DESCRIPTION
**Pull request by tautschnig**
Just as previously done for core: this will enable future use of `safety::{ensures,requires}` in those crates.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: library/std/src/lib.rs [hunk 1]
```diff
@@ -269,6 +269,7 @@
 #![cfg_attr(any(windows, target_os = "uefi"), feature(round_char_boundary))]
 #![cfg_attr(target_family = "wasm", feature(stdarch_wasm_atomic_wait))]
 #![cfg_attr(target_arch = "wasm64", feature(simd_wasm64))]
+#![cfg_attr(kani, feature(kani))]
```
[the remark this answers is not in the document]
Comment (tautschnig) [reply library/std/src/lib.rs:272]: I don't know what the optimal solution is, but I was running into `error[E0658]: use of unstable library feature 'kani'` so I followed cargo's guidance which said ```add `#![feature(kani)]` to the crate attributes to enable```.

[the remark this answers is not in the document]
Comment (celinval) [reply library/std/src/lib.rs:272]: The reason is that we add APIs under unstable kani feature. So to use them in other crates, you need to define the feature

[the remark this answers is not in the document]
Comment (tautschnig) [reply library/std/src/lib.rs:272]: So ... is what I am proposing here ok, or is there another way it should be done?

[the remark this answers is not in the document]
Comment (celinval) [reply library/std/src/lib.rs:272]: What you are proposing here is correct.

**Review by carolynzech — APPROVED** [review]

File: library/std/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (celinval) [reply library/std/src/lib.rs:272]: I think this PR is missing context... the reason why you will need this feature s to invoke `core::kani` module functions, not `requires` and `ensures`. Note that in https://github.com/model-checking/verify-rust-std/pull/43, we already added contracts to `alloc`.

**Review by qinheping — APPROVED** [review]
LGTM

File: library/std/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tautschnig) [reply library/std/src/lib.rs:272]: That was `core/src/alloc`, not the `alloc` crate that lives next to `core`.

**Review by zhassan-aws — APPROVED** [review]

**Pull request merged**